geomstats/geometry/hyperboloid.py

```diff
@@ -0,0 +1,427 @@
+
+        sq_norm = self.embedding_metric.squared_norm(base_point)
+        inner_prod = self.embedding_metric.inner_product(base_point,
+                                                         vector)
+
+        coef = inner_prod / sq_norm
+
+        tangent_vec = vector - gs.einsum('...i,...j->...j', coef, base_point)
+        return tangent_vec
+
+    def intrinsic_to_extrinsic_coords(self, point_intrinsic):
+        """Convert from intrinsic to extrinsic coordinates.
+
+        Parameters
+        ----------
+        point_intrinsic : array-like, shape=[n_samples, dim]
+            Point in the embedded manifold in intrinsic coordinates.
+        """
+        return\
+            Hyperbolic.change_coordinates_system(point_intrinsic,
+                                                 'intrinsic',
+                                                 'extrinsic')
+
+    def extrinsic_to_intrinsic_coords(self, point_intrinsic):
+        """Convert from extrinsic to intrinsic coordinates.
+
+        Parameters
+        ----------
+        point_extrinsic : array-like, shape=[n_samples, dim_embedding]
+            Point in the embedded manifold in extrinsic coordinates,
+            i. e. in the coordinates of the embedding manifold.
+        """
+        return\
+            Hyperbolic.change_coordinates_system(point_intrinsic,
+                                                 'extrinsic',
+                                                 'intrinsic')
+
+
+class HyperboloidMetric(HyperbolicMetric):
+    """Class that defines operations using a hyperbolic metric.
+
+    Parameters
+    ----------
+    dimension : int
+        Dimension of the hyperbolic space.
+    point_type : str, {'extrinsic', 'intrinsic', etc}, optional
+        Default coordinates to represent points in hyperbolic space.
+    scale : int, optional
+        Scale of the hyperbolic space, defined as the set of points
+        in Minkowski space whose squared norm is equal to -scale.
+    """
+
+    default_point_type = 'vector'
+    default_coords_type = 'extrinsic'
+
+    def __init__(self, dimension, coords_type='extrinsic', scale=1):
+        super(HyperboloidMetric, self).__init__(
+            dimension=dimension,
+            scale=scale)
+        self.embedding_metric = MinkowskiMetric(dimension + 1)
+
+        self.coords_type = coords_type
+        self.point_type = HyperbolicMetric.default_point_type
+
+        self.scale = scale
+
+    def inner_product_matrix(self, base_point=None):
+        """Compute the inner product matrix, independent of the base point.
+
+        Parameters
+        ----------
+        base_point: array-like, shape=[n_samples, dimension]
+
+        Returns
+        -------
+        inner_prod_mat: array-like, shape=[n_samples, dimension, dimension]
+        """
+        self.embedding_metric.inner_product_matrix(base_point)
+
+    def _inner_product(self, tangent_vec_a, tangent_vec_b, base_point=None):
```

Comment:
Why is this a private method? Also, can it be used in conjunction with the scaling of inner_product and norm from the parent class?